osu.Game/Screens/Edit/Screens/Setup/Components/SwitchButton.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Configuration;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.States;
+using osu.Game.Graphics;
+
+namespace osu.Game.Screens.Edit.Screens.Setup.Components
+{
+    public class SwitchButton : CompositeDrawable, IHasCurrentValue<bool>, IHasAccentColour
+    {
+        private readonly Box fill;
+        private readonly Container switchContainer;
+        private readonly CircularContainer circularContainer;
+
+        private const float border_thickness = 4.5f;
+        private const float switch_padding = 1.25f;
+        private const float size_x = 45;
```

Comment:
not much point having this here if it's only used in one place. also we generally call this "width".